Library/Homebrew/test/test_integration_cmds.rb

```diff
@@ -757,6 +757,47 @@ def test_analytics
     assert_match "Analytics is disabled", cmd("analytics")
   end
 
+  def test_migrate
+    setup_test_formula "testball1"
+    setup_test_formula "testball2"
+    assert_match "Invalid usage", cmd_fail("migrate")
+    assert_match "No available formula with the name \"testball\"",
+      cmd_fail("migrate", "testball")
+    assert_match "testball1 doesn't replace any formula",
+      cmd_fail("migrate", "testball1")
+
+    core_tap = CoreTap.new
+    core_tap.path.cd do
+      shutup do
+        system "git", "init"
+        system "git", "add", "--all"
+        system "git", "commit", "-m", "Testball1 has not yet been renamed"
+      end
+    end
+
+    cmd("install", "testball1")
+    (core_tap.path/"Formula/testball1.rb").unlink
+    formula_renames = core_tap.path/"formula_renames.json"
+    formula_renames.write <<-EOS.undent
+    {
+      "testball1": "testball2"
+    }
+    EOS
+
+    core_tap.path.cd do
+      shutup do
+        system "git", "add", "--all"
+        system "git", "commit", "-m", "Testball1 has been renamed to Testball2"
+      end
+    end
+
+    assert_match "Migrating testball1 to testball2", cmd("migrate", "testball1")
+    (HOMEBREW_CELLAR/"testball1").unlink
+    assert_match "Error: No such keg", cmd_fail("migrate", "testball1")
+  ensure
+    formula_renames.unlink
```

Comment:
Better to add this to the `rm_rf` list above otherwise if this fails before `formula_renames` is created the `ensure` will fail.